specs/validator/tableland-openapi-spec.yaml

```diff
@@ -116,6 +118,10 @@ paths:
                 type: object
         "400":
           description: Invalid query/statement value
+        "404":
+          description: Row Not Found
```

Comment:
So this `404` is a side-effect of the current (legacy) APIs, so it isn't an intentional decision of the new one.

IMHO, I never understood why we make an exception if the query results in no rows instead of returning `[]`.
Like, a query not returning rows doesn't seem to me to be something "not ok"... depending on the case, you might even _expect_ that a table might not have rows.

I always made sense of this because maybe it was useful for some particular case out there (maybe marketplaces?), where they'd prefer a 404 instead of a 200 with an empty result.

I don't have a strong personal opinion, both "200 + empty `[]`" or "404" are fine to me, but this is a good opportunity to surface it.

Reply:
Probably the best approach here is just leaving the 404, since I'm pretty sure has a good reason.

Also, I think that if we don't have rows, it becomes more complicated to figure out which would be the columns of a possible non-empty result (since columns names depends on each read-query). Actually, that sounds like a pretty complex problem and may be the underlying reason for this original decision.

Reply:
Yeh I agree with you. It was actually a pain to deal with this behavior, as this is entirely expected results. Dealing with it as an error seems silly. Perhaps @andrewxhill or @joewagner has some insight here? Either way, seems like we'll need to keep it for now until we discover otherwise?

Reply:
As a general opinion, It makes intuitive sense to me to return "200 + empty `[]`" if the table exists and is empty, and return 404 if the table doesn't exist.
I don't know of anything that would be broken by this change, but it seems like a breaking change so implementing it as part of `/api/v1` also makes sense to me.

Reply:
yeah... agree all around. nothing i've built would be impacted here

Reply:
> As a general opinion, It makes intuitive sense to me to return "200 + empty `[]`" if the table exists and is empty, and return 404 if the table doesn't exist. 

@joewagner, note that this is the `/query` API, so actually we aren't in a case where we can say "a table exists" or doesn't. This is related to a read-query that results in no rows (e.g: `SELECT * FROM foo_1_2 WHERE 1=2` where `foo_1_2` exists).

---
@carsonfarmer @andrewxhill @joewagner, thanks for providing more signal if this might be strictly needed or not.

I tested doing some changes to remove this 404, and I discarded what I feared bout not being able to have the column names, so: TL;DR, I discard the theory that this might be motivated by some "technical" challenge around that; no problem with that.

Since this 404 was originally included by @sanderpick, maybe it's worth waiting for his opinion. My only fear is that this is needed for some marketplace reasons that I (we) don't know about... if it's needed by some application out there, I think it makes sense to tell them that 404 is gone.

Considering we'll ship this by ~January, maybe we have a bit more time to see if @sanderpick might read this and provide his opinion (I wouldn't vote to ping him directly since he's OOO, but I'm seeing him post things once in a while in GH so maybe we can give him some time). 

Depending on that, we can make a call in ~2 weeks. From the validator side, it's a doable change (I still have to provide the same behavior to the old JSON-RPC but the new one for API V1, since they share code), and I'd expect from the SDK side it's removing a particular case that is handled today.

@carsonfarmer, wdyt?

Reply:
yeh I think that's a perfect plan! let's make the call in ~2 weeks with a separate PR as needed.